ee/server/service/users.go

```diff
@@ -61,3 +101,55 @@ func (svc *Service) GetSSOUser(ctx context.Context, auth fleet.Auth) (*fleet.Use
 	}
 	return user, nil
 }
+
+// rolesChanged checks whether there was any change between the old and new roles.
+//
+// rolesChanged assumes that there cannot be multiple role entries for the same team.
+func rolesChanged(oldGlobal *string, oldTeams []fleet.UserTeam, newGlobal *string, newTeams []fleet.UserTeam) bool {
+	if (newGlobal != nil && (oldGlobal == nil || *oldGlobal != *newGlobal)) || (newGlobal == nil && oldGlobal != nil) {
+		return true
+	}
+	if len(oldTeams) != len(newTeams) {
+		return true
+	}
+	oldTeamsMap := make(map[uint]fleet.UserTeam)
```

Comment:
Wouldn't `reflect.DeepEqual(a, b)` work here?

Reply:
I just want to compare the team `ID` and the `Role`, `fleet.UserTeam` has more fields (that `reflect.DeepEqual` would use to compare) and we might add new fields in the future that may break the logic if we use `reflect.DeepEqual`.
```
// In general DeepEqual is a recursive relaxation of Go's == operator.
// However, this idea is impossible to implement without some inconsistency.
// Specifically, it is possible for a value to be unequal to itself,
// either because it is of func type (uncomparable in general)
// or because it is a floating-point NaN value (not equal to itself in floating-point comparison),
// or because it is an array, struct, or interface containing
// such a value.
```
Though am all ears if we believe there's a good reason to use `reflect.DeepEqual`.

Reply:
SGTM.